src/js/components/Search/SelectOption.js

```diff
@@ -1,20 +1,33 @@
-import React, { useEffect, useState } from 'react';
+import React, { useRef, useEffect, useState } from 'react';
 import { Select } from 'antd';
 import { useDispatch } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 
 import { addQueryParam } from '../../features/search/query';
 
-const SelectOption = ({ id, isChecked, options }) => {
+const SelectOption = ({ id, isChecked, options, optionalDispatch}) => {
   const { t } = useTranslation();
   const dispatch = useDispatch();
 
   const [value, setValue] = useState(options[0]);
 
+  const isFirstRun = useRef(true);
   useEffect(() => {
     if (isChecked) {
       dispatch(addQueryParam({ id, value }));
```

Comment:
If the user un-checks this option, won't the query param be out of sync?

Reply:
Nope
This 'first-run' is only triggered upon first page and component load